libs/components/src/next/form/checkbox/vl-checkbox.component.ts

```diff
@@ -0,0 +1,162 @@
+import { checkboxStyle } from '@domg/govflanders-style/component';
+import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
+import { CSSResult, html, PropertyDeclarations, TemplateResult } from 'lit';
+import { customElement } from 'lit/decorators.js';
+import { vlElementsStyle } from '@domg-wc/elements';
+import { classMap } from 'lit/directives/class-map.js';
+import checkboxUigStyle from './vl-checkbox.component.uig-css';
+import { FormControl } from '../form-control/FormControl';
+
+export const CheckboxDefaults = {
+    value: '',
+    block: false,
+    isSwitch: false,
+    checked: false,
+    disabled: false,
+    error: false,
+    label: 'label',
+    name: '',
+    value: '',
+};
+
+@customElement('vl-checkbox-next')
+export class VlCheckboxComponent extends FormControl {
+    private value = CheckboxDefaults.value;
+    private checked = CheckboxDefaults.checked;
+    private isSwitch = CheckboxDefaults.isSwitch;
+    private initialValue: string | boolean = '';
+    private initialCheckedValue: boolean | undefined = undefined;
+    private touched = false;
+    static get styles(): (CSSResult | CSSResult[])[] {
+        // TODO: why do we need to add vlElementsStyle here?
+        return [resetStyle, baseStyle, vlElementsStyle, checkboxStyle, checkboxUigStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            checked: { type: Boolean, reflect: true },
+            value: { type: String, attribute: 'value', reflect: true },
+            isSwitch: { type: Boolean, attribute: 'switch', reflect: true },
+        };
+    }
+
+    constructor() {
+        super();
+
+        this.disabled = CheckboxDefaults.disabled;
+        this.label = CheckboxDefaults.label;
+        this.block = CheckboxDefaults.block;
+        this.error = CheckboxDefaults.error;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+
+        if (!this.initialValue) {
+            this.initialValue = this.value;
+            if (this.checked) this.initialCheckedValue = this.checked;
+        }
+    }
+
+    updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+
+        if (changedProperties.has('checked') && changedProperties.get('checked') !== undefined) {
+            if (this.checked) {
+                this.setValue(this.value || 'true');
+            } else {
+                this.setValue('');
+            }
+            if (this.touched || this.initialCheckedValue !== this.checked) {
+                this.checkValidity();
```

Comment:
Hier zou ik mee oppassen, het is de bedoeling dat er enkel gevalideerd wordt bij het submitten.

Reply:
De forms die ik vroeger opzette en persoonlijk ook handig vind in gebruik, wordt er bij input ook al meteen gevalideerd. Hiermee wil ik niet zeggen dat het zo moet bij ons.

In mijn mening, gaat submit, naast submitten zelf, ook valideren maar idealiter weet de gebruiker tegen dan al of de data die ingevuld is, valid is.

Als je hier niet van overtuigd bent, dan pas ik dit aan, en kunnen we mogelijks later een soort van `reactive` optie toevoegen bij de FormControl's, waarbij er al meteen gevalideerd wordt na input?

Reply:
zoals besproken; voorlopig enkel validatie bij submit - later herbekijken we reactiviteit